### ISSUE TITLE
Standalone JS pkg

### ISSUE THREAD

**Issue by manzt** [manzt/quak#16]
The custom DataTable Mosaic client should be published to npm and usable without the Python client.

**Comment by timelyportfolio** [comment]
@manzt Absolutely loving `quak` and would really like to see it available as a standalone JS package.  If you pursue this option, would it be possible to add an option `filterBy` similar to `mosaic`/`vgplot` so that we can tie together `DataTable` with `mosaic` inputs?  Here is a really, really ugly example (please excuse my foolish ignorance of Typescript and lots of other things) https://stackblitz.com/~/github.com/timelyportfolio/quak.

**Comment by domoritz** [comment]
In other words, make quak a mosaic component that satisfies the component interface?

**Comment by timelyportfolio** [comment]
@domoritz yes, would be a really nice addition to the components.

**Comment by timelyportfolio** [comment]
And while I originally was only thinking the `DataTable` it would also be even nicer if each of the filters in the column headings would be available as well as `input` components.

**Comment by manzt** [comment]
Yes! This is my intention. I _think_ `DataTable` is a mosaic component (it extends from MosaicClient). Right now you could access the `filterBy` with:

```javascript
import { coordinator } from "@uwdata/mosaic-core";
import { datatable } from "./lib/clients/DataTable.ts";

// setup df table in database...

const dt = await datatable("df", { coordinator: coordinator() });
const filterBy = dt.filterBy; // use with other clients
```

Originally I'd had this lifted up one level, where you could create and pass in the crossfilter selection (`filterBy`) yourself, but I don't see much of a difference between the APIs (and also potential to pass in a Selection that wasn't a crossfilter felt like a poor API design). Open to reverting to:

```javascript
import { coordinator, Selection } from "@uwdata/mosaic-core";
import { datatable } from "./lib/clients/DataTable.ts";

// setup df table in database...
const filterBy = Selection.crossfilter();
const dt = await datatable("df", { coordinator: coordinator(), filterBy: filterBy });
```

if there are things I'm missing!

**Comment by manzt** [comment]
> Here is a really, really ugly example (please excuse my foolish ignorance of Typescript and lots of other things) stackblitz.com/~/github.com/timelyportfolio/quak.

wow, totally love that we can share ideas with stackblitz this way. thanks for sharing, @timelyportfolio  :)

**Comment by aszenz** [comment]
@manzt Is this completed, how can i install the JS pkg?
